addons/account/models/account_move.py

```diff
@@ -5049,6 +5049,18 @@ def _disable_recursion(self, container, key, default=None, target=True):
     def _mailing_get_default_domain(self, mailing):
         return ['&', ('move_type', '=', 'out_invoice'), ('state', '=', 'posted')]
 
+    @api.model
+    def _routing_check_route(self, message, message_dict, route, raise_exception=True):
+        if ((route and route[2]) or {}).get('move_type', 'entry') in ('out_invoice', 'in_invoice') and len(message_dict['attachments']) < 1:
```

Comment:
```suggestion
        if route[0] == 'account.move' and len(message_dict['attachments']) < 1:
```
We can be more generic here, and catch all emails that are targeting `account.move` without attachment (checked with @chklop)